### ISSUE TITLE
No working UI, and no config file created

### ISSUE THREAD

**Issue by dln949** [gerbera/gerbera#410]
I have a fresh installation of Gerbera 1.2.0 from the Czetty PPA on Ubuntu Bionic 18.04.1.  

The instructions at http://docs.gerbera.io/en/latest/config-overview.html#generating-configuration say that at first run, to run the command $ gerbera --create-config .  While that put output into the terminal, it did not create an xml file in ~/.config/gerbera.  So I went ahead with the command gerbera --create-config | sudo tee /etc/gerbera/config.xml , which did create the xml file in /etc/gerbera.  So, I copied that xml file to ~/.config/gerbera.  Here are the permissions for ~/.config/gerbera, if that helps:  

main1@system1:~/.config/gerbera$ ls -l config.xml
-rw-r--r-- 1 gerbera gerbera 6739 Jan  8 22:50 config.xml

When I start gerbera, it goes to this URL:  http://192.168.1.5:49152/disabled.html .  I get a message that says, "Gerbera's UI is disabled.  If you did not expect this, please check your configuration."  

When I remove the /disabled.html part of the URL, I get a screen that "looks" like things are going to work, but.....  

At the bottom of the screen is a popup with the message, : "The UI is disabled in the configuration file.  See README."  And, I get another message that says, "The system encountered an error", with a sad face emoticon.  The only "clickable" things that work are the "Documentation" and the "Report an Issue" links.  If I click on  "Login", I am taken to a page that says, "This site can't be reached", and "ERR_CONNECTION_REFUSED".  

Here is what is in both config files for the server ui:  

<server>
    <ui enabled="yes" show-tooltips="yes">
      <accounts enabled="no" session-timeout="30">
        <account user="gerbera" password="gerbera"/>
      </accounts>
    </ui>

Soooooo.....  

1)  Why would the ~/.config/gerbera/config.xml file not be created when I ran the command  $ gerbera --create-config ?  

2)  Why do I need two config files in two places?  

3)  Why do I get the error about the UI being disabled in the configuration file(s) when they both clearly have the server ui set to enabled ("yes")?  (I am aware of issue 319 here https://github.com/gerbera/gerbera/issues/319 , but that issue didn't really tell me what I ought to do in my case to solve this problem.)  

4)  Why does clicking on "Login" take me to that screen about refused connections, instead of to a login screen?

**Comment by v00d00** [comment]
You broadly have two options:

1. Run gerbera as system service (recommended)

When you run gerbera as service, it will load its configuration from `/etc/gerbera/config.xml` ([systemd unit](https://github.com/gerbera/gerbera/blob/master/scripts/systemd/gerbera.service.cmake)) and be run as the gerbera user.

To run it using systemd: `systemctl start gerbera`.

2. Run it under your user

This is what happens when you "gerbera" in a shell. It will look for configuration in `~/.config/gerbera`. It runs as your user like any other. `~/.config/gerbera` Should belong to you.

**Comment by v00d00** [comment]
> Why would the ~/.config/gerbera/config.xml file not be created when I ran the command $ gerbera --create-config ?

As the docs mention, it only outputs the content to standard output. It does not create any config files. @elmodaddyb I guess we should rename this, but we would need to keep the old option for a while too. perhaps with a warning?

>  Why do I need two config files in two places?

You dont. It depends how you want to use it, see my post above.

> Why do I get the error about the UI being disabled in the configuration file(s) when they both clearly have the server ui set to enabled ("yes")? (I am aware of issue 319 here #319 , but that issue didn't really tell me what I ought to do in my case to solve this problem.)
>  Why does clicking on "Login" take me to that screen about refused connections, instead of to a login screen?

This happens when the server fails to start for some reason, are you sure Gerbera is running?

**Comment by elmodaddyb** [comment]
Yea, i guess `--create-config` is misleading since it does not **create** anything.

Maybe:

`gerbera --example-config`
or
`gerbera --generate-config`

**Comment by dln949** [comment]
No, I am not sure that the gerbera service is running.  How can I be sure.  

I started gerbera per the docs using this command:  gerbera -c /etc/gerbera/config.xml .  

Also, I started gerbera from the activities menu in ubuntu.  Is tgere some other way i am supposed to start it?

**Comment by v00d00** [comment]
As I mentioned the recommended way is like you would for any other service:

`sudo systemctl start gerbera.`

**Comment by dln949** [comment]
Ok, let me see if I am understanding this correctly......  

Using gerbera is a two step process:  

1) start the gerbera service; and then, 
2) start gerbera (e.g., from the activities menu)  

Is that correct?  

I have never had to manually start a service for any other application, so I did not understand that is required in this case.  

Is there a configuration option within gerbera to start gerbera automatically when the computer starts?

**Comment by dln949** [comment]
Ok, I booted up the computer, checked the status of gerbera service, it was not running.  Then:  


main1@system1:~$ systemctl start gerbera
main1@system1:~$ systemctl status gerbera
● gerbera.service - Gerbera Media Server
   Loaded: loaded (/lib/systemd/system/gerbera.service; disabled; vendor preset:
   Active: activating (auto-restart) (Result: exit-code) since Wed 2019-01-09 17
     Docs: man:gerbera(1)
           https://gerbera.io/
  Process: 15083 ExecStart=/usr/bin/gerbera -c /etc/gerbera/config.xml $EXTRAARG
 Main PID: 15083 (code=exited, status=1/FAILURE) (code=exited, status=1/FAILURE)
lines 1-7/7 (END)


main1@system1:~$ journalctl -u gerbera
Jan 08 23:20:12 system1 systemd[1]: Started Gerbera Media Server.
Jan 08 23:20:12 system1 gerbera[6633]: 2019-01-08 23:20:12    INFO: Gerbera UPnP Server version 1.2.0 - http://gerbera.io/
Jan 08 23:20:12 system1 gerbera[6633]: 2019-01-08 23:20:12    INFO: ===============================================================================
Jan 08 23:20:12 system1 gerbera[6633]: 2019-01-08 23:20:12    INFO: Gerbera is free software, covered by the GNU General Public License version 2
Jan 08 23:20:12 system1 gerbera[6633]: 2019-01-08 23:20:12    INFO: Copyright 2016-2018 Gerbera Contributors.
Jan 08 23:20:12 system1 gerbera[6633]: 2019-01-08 23:20:12    INFO: Gerbera is based on MediaTomb: Copyright 2005-2010 Gena Batsyan, Sergey Bostandzhyan, Leonhard 
Jan 08 23:20:12 system1 gerbera[6633]: 2019-01-08 23:20:12    INFO: ===============================================================================


I do not know how to interpret the above results to determine what is wrong, why the service appears not to run.  Can you help?  (Also - the dates and times in the journal appear all wrong.  They show Jan 8, 23:20, but it is now Jan 9, about 17:50 on my computer.)

**Comment by jkiusAlarus** [comment]
hello
I is possible that running gerbera --create-config | sudo tee /etc/gerbera/config.xml overwrited the system config with a local user user one => that coild explain why gerbera tries to open your /home/main1/.config

what is the content of the <home> tag in /etc/gerbera/config.xml? It should be  /etc/gerbera/ and not 
/home/main1/

**Comment by dln949** [comment]
> hello
> It is possible that running gerbera --create-config | sudo tee /etc/gerbera/config.xml overwrited the system config with a local user user one => that could explain why gerbera tries to open your /home/main1/.config
> 
> what is the content of the tag in /etc/gerbera/config.xml? It should be /etc/gerbera/ and not
> /home/main1/

Thank you, but....  I'm sorry.  What is "the tag" that you are asking about?  When I open that xml file, there are many tags.  I don't know which you are referring to.  Or....  Is there supposed to be only one tag in that file?  Because if that is true, then I fear that I have a much larger problem than I imagined.

**Comment by jkiusAlarus** [comment]
sorry, I meant the "home" tag. It got skipped because I wrote it like an html tag

**Comment by dln949** [comment]
> sorry, I meant the "home" tag. It got skipped because I wrote it like an html tag

Thank you!  That worked.  Well, it was one of two things that I had to do.  

As you suggested, I had to change the path in the "home" key in the xml file to /etc/gerbera/.  But, I then also noticed that the ownership of /etc/gerbera and the config file within it was "root", and not "gerbera".  (I have no idea how that happened.  I do not think that _I_ did that.  But then, if I did not do that - Why would gerbera not put the correct ownership on that?  Or, why wouldn't the documentation warn that it needs to be changed?)  So, I also had to make that change.  And now.....  It all works!  

Thank you again for the help.  

In hindsight, I can see what went wrong.  To a laymen computer person like me, I simply attempted to follow all the steps shown in the documentation for installation/configuration.  Now I see that was a mistake.  (Although I think that documentation does not make clear which commands should be used in which circumstance.)
